op-supervisor/supervisor/backend/backend.go

```diff
@@ -423,94 +423,94 @@ func (su *SupervisorBackend) DependencySet() depset.DependencySet {
 // Query methods
 // ----------------------------
 
-func (su *SupervisorBackend) CheckMessage(ctx context.Context, identifier types.Identifier, payloadHash common.Hash, executingDescriptor types.ExecutingDescriptor) (types.SafetyLevel, error) {
-	logHash := types.PayloadHashToLogHash(payloadHash, identifier.Origin)
-	chainID := identifier.ChainID
-	blockNum := identifier.BlockNumber
-	logIdx := identifier.LogIndex
-	_, err := su.chainDBs.Contains(chainID,
-		types.ContainsQuery{
-			BlockNum:  blockNum,
-			Timestamp: identifier.Timestamp,
-			LogIdx:    logIdx,
-			LogHash:   logHash,
-		})
-	if errors.Is(err, types.ErrFuture) {
-		su.logger.Debug("Future message", "identifier", identifier, "payloadHash", payloadHash, "err", err)
-		return types.LocalUnsafe, nil
+func (su *SupervisorBackend) checkAccess(ctx context.Context,
```

Comment:
I find it a little odd that `checkAccess` doesn't have any notion of Minimum Safety, while `CheckAccessList` does. Could we delegate the Min-Safety check down to this function so that the `.*List` function is really just a wrapper around the individual one?

Reply:
I thought about that, even intermittently tried it, but the safety-level depends on block-content. There are a few cheks related to rewind-locking. Once we resolve those, we can make the supervisor only check the *latest* block per involved chain, instead of all of them. This can save DB reads for longer access-lists. So hence I don't think it should be tightly coupled to the individual access checks.